website/content/docs/job-specification/service.mdx

```diff
@@ -173,6 +173,17 @@ Connect][connect] integration.
   `meta` parameter. If this is not supplied, the registered meta will be set to
   that of the `meta` parameter.
 
+- `on_update` `(string: "require_healthy")` - Specifies how checks should be
+  evaluated when determining deployment health.
+
+  - `require_healthy` - In order for Nomad to consider the check healthy during
```

Comment:
I think this is a bigger feature than we're letting on by this pithy description. Let's both tighten up our use of terminology here and expand this to cover some of the things discussed in the RFC. It's worth over-explaining what behaviors to expect.

* Why does a job submitter want to use this?
* Does this happen on a deployment (which is specific to only `service` jobs) or on a job update? (Which, come to think of it... should this have been called `on_deployment` if that's the case?)
* What about the first time the job is deployed?
* Is it only that the check isn't being considered healthy if its not healthy (which seems tautological) or is it that the deployment won't be marked healthy?
* What's happening in Consul with checks in these states?

Reply:
Good points I'll update, few comments / questions inline

- Services only apply to `service` jobs anyways right? In the future if Nomad ever does checks outside the scope of consul I think `on_update` will still make sense
- It applies only to a service's checks, but the overall deployment is considered unhealthy/healthy based on the health of all the checks

Reply:
Services can also be applied to `system` jobs, as you may need them to be service-discoverable. You can even use them with `batch` jobs as well, although it's... well that's just a weird thing to do! 😀